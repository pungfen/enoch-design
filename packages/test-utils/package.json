{
  "name": "@enochfe/test-utils",
  "author": "pung <fp94214@gmail.com>",
  "description": "enoch test-utils",
  "version": "1.0.1",
  "type": "module",
  "main": "src/index.ts",
  "scripts": {
    "build": "tsup"
  },
  "dependencies": {
    "@vitejs/plugin-vue-jsx": "^2.0.1",
    "esbuild": "^0.15.10",
    "rollup": "^3.1.0",
    "rollup-plugin-esbuild": "^4.10.1",
    "unplugin": "^0.9.6",
    "unplugin-vue": "3.1.3"
  },
  "devDependencies": {
    "vite": "^3.1.3",
    "vue": "^3.2.39"
  }
}
