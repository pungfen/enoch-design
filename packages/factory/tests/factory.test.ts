import { mount } from '@vue/test-utils'

import FactoryBasic from './factory/factory.basic.vue'

describe('factory basic', () => {
  test('mount component', () => {
    expect(FactoryBasic).toBeTruthy()
  })

  test(`shoule render 'form data name' correctly`, () => {
    const wrapper = mount(FactoryBasic)
    const formElement = wrapper.find('.form-data-name')
    expect(formElement.text()).toBe('pung')
    expect(wrapper.html()).toMatchSnapshot()
  })

  // test(`shoule render 'form data id' correctly`, () => {
  //   const wrapper = mount(FactoryBasic)
  //   const formElement = wrapper.find('.form-data-id')
  //   expect(formElement.text()).toBe('0')
  //   expect(wrapper.html()).toMatchSnapshot()
  // })

  // test(`emit 'header.add.click' event, shoule render 'form data id' correctly`, async () => {
  //   const wrapper = mount(FactoryBasic)
  //   const buttonElement = wrapper.get('button')
  //   const formElement = wrapper.find('.form-data-id')
  //   expect(formElement.text()).toContain('0')

  //   await buttonElement.trigger('click')

  //   expect(formElement.text()).toContain('1')

  //   expect(wrapper.html()).toMatchSnapshot()
  // })

  // test(`should render 'form.fullname' computed correctly`, () => {
  //   const wrapper = mount(FactoryBasic)
  //   const fullname = wrapper.find('.form-fullname')
  //   const name = wrapper.find('.form-data-name')
  //   expect(fullname.text()).toContain('我叫' + name.text())
  //   expect(fullname.html()).toMatchSnapshot()
  // })
})
