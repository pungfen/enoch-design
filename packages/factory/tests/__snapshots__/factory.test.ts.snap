// Vitest Snapshot v1

exports[`factory basic > emit 'header.add.click' event, shoule render 'form data id' correctly 1`] = `
"<button>+</button>
<div class=\\"form-data-id\\">1</div>
<div class=\\"form-data-name\\">pung</div>
<div class=\\"form-fullname\\">我叫pung</div>"
`;

exports[`factory basic > should render 'form.fullname' computed correctly 1`] = `"<div class=\\"form-fullname\\">我叫pung</div>"`;

exports[`factory basic > shoule render 'form data id' correctly 1`] = `
"<button>+</button>
<div class=\\"form-data-id\\">0</div>
<div class=\\"form-data-name\\">pung</div>
<div class=\\"form-fullname\\">我叫pung</div>"
`;

exports[`factory basic > shoule render 'form data name' correctly 1`] = `
"<button>+</button>
<div class=\\"form-data-id\\">0</div>
<div class=\\"form-data-name\\">pung</div>
<div class=\\"form-fullname\\">我叫pung</div>"
`;
